src/eedl/zonal.py

```diff
@@ -27,8 +27,8 @@ def zonal_stats(polygons: Union[str, Path],
 				raster: Union[str, Path],
 				output_folder: Union[str, Path],
 				filename: str,
-				keep_fields: Tuple[str] = ("UniqueID", "CLASS2"),
-				stats: Tuple[str] = ('min', 'max', 'mean', 'median', 'std', 'count', 'percentile_10', 'percentile_90'),
+				keep_fields: Tuple[str, str] = ("UniqueID", "CLASS2"),
+				stats: Tuple[str, str, str, str, str, str, str, str] = ('min', 'max', 'mean', 'median', 'std', 'count', 'percentile_10', 'percentile_90'),
```

Comment:
Both this tuple and the previous one could be shorter or longer. They'll both be an iterable (not just a tuple, but possibly a list too() of strings, but we don't know how long they'll be. Can we get this converted so that it is a Tuple of strings with length *? `stats` will be one or more, but keep_fields isn't guaranteed to have a length larger than 0 (it can be valid with length 0). Let me know if I'm not making sense.

Reply:
On line 65 there is an addition between keep_fields and stats. If we expect any iterable (1 could be a list and the other a tuple), how do you want to handle this addition?
`fieldnames = stats + keep_fields`

Reply:
Good point. I'm fine with it being a tuple for the sake of our internal type checking and if people at runtime want to provide something else, it's on them to make sure the type mix is compatible.